corehq/apps/cleanup/management/commands/fix_forms_with_missing_xmlns.py

```diff
@@ -0,0 +1,175 @@
+import uuid
+
+from couchdbkit import ResourceNotFound
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.app_manager.util import get_correct_app_class
+from corehq.apps.app_manager.xform import XForm, parse_xml
+from corehq.apps.es import FormES
+from corehq.apps.es.filters import NOT
+from corehq.apps.es.forms import xmlns
+from couchforms.const import ATTACHMENT_NAME
+from couchforms.models import XFormInstance
+from dimagi.utils.couch.database import iter_docs
+from django.core.management.base import BaseCommand
+from optparse import make_option
+
+
+class Command(BaseCommand):
+    help = 'Fix forms with "undefined" xmlns'
+    args = '<log_path>'
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help="Don't do the actual modifications, but still log what would be affected"
+        ),
+    )
+
+    def handle(self, *args, **options):
+
+        dry_run = options.get("dry_run", True)
+        log_path = args[0].strip()
+
+        new_xmlnss = {}
+        num_fixed = 0
+        total, submissions = get_submissions_without_xmlns()
+
+        with open(log_path, "w") as f:
+            for i, xform_instance in enumerate(submissions):
+                if i % 500 == 0 and i != 0:
+                    print "Progress: {} of {} ({})".format(i, total, round(i / float(total), 2))
+                    print "We can fix {} of {} ({})".format(
+                        num_fixed, i, round(num_fixed / float(i), 2)
+                    )
+                if xform_instance.app_id in new_xmlnss:
+                    num_fixed += 1
+                    # We've already generated a new xmlns for this app
+                    set_xmlns_on_submission(xform_instance, new_xmlnss[xform_instance.app_id], dry_run)
+                    f.write("Set new xmlns on submission {}\n".format(xform_instance._id))
+                else:
+                    app = Application.get(xform_instance.app_id)
+                    forms_without_xmlns = get_forms_without_xmlns(app)
+                    if len(forms_without_xmlns) == 1:
+                        form = forms_without_xmlns[0]
+                        if not xforms_with_real_xmlns_possibly_exist(app._id, form):
```

Comment:
Should have been more explicit about that, sorry. Currently not planning on attempting to fix forms where `xforms_with_real_xmlns_possibly_exist()` is true, because there could be false positives, and I haven't thought through how exactly to determine if these form matches are real or not. I'll either try to match by questions, or even look at them manually if the number is low enough.